api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java

```diff
@@ -0,0 +1,1127 @@
+ /**
+  * This Source Code Form is subject to the terms of the Mozilla Public License,
+  * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+  * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+  * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+  * <p>
+  * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+  * graphic logo is a trademark of OpenMRS Inc.
+  */
+
+package org.openmrs.module.locationbasedaccess.aop;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Cohort;
+import org.openmrs.Encounter;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.User;
+import org.openmrs.Visit;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.VisitService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.locationbasedaccess.aop.common.AOPContextSensitiveTest;
+import org.openmrs.module.locationbasedaccess.aop.common.TestWithAOP;
+import org.openmrs.module.locationbasedaccess.aop.interceptor.UserServiceInterceptorAdvice;
+import org.openmrs.parameter.EncounterSearchCriteria;
+
+public class EncounterSearchAdviserTest extends AOPContextSensitiveTest {
+    
+    private EncounterService encounterService;
+    private UserService userService;
+    private LocationService locationService;
+    private PatientService patientService;
+    private VisitService visitService;
+    private static final String XML_FILENAME_WITH_ADMIN_USER_DATA = "include/AdminUserData.xml";
+    private static final String XML_FILENAME_WITH_PERSON_DATA = "include/PatientTestData.xml";
+    
+    private static final int DEMO_USER1_ID = 6001;
+    private static final int DEMO_USER2_ID = 6002;
+    private static final int DEMO_LOCATION1_ID = 1;
+    private static final int DEMO_LOCATION2_ID = 2;
+    
+    public EncounterSearchAdviserTest() {
+    }
+    
+    @Override
+    protected void setInterceptorAndServices(TestWithAOP testCase) {
+        testCase.setInterceptor(UserServiceInterceptorAdvice.class);
+        testCase.addService(UserService.class);

```

Comment:
You have connected with UserServiceInterceptorAdvice, UserService classes here, then how can you test for EncounterServiceInterceptorAdvice and EncounterService?

Reply:
Made the necessary changes 😃

Reply:
Can you confirm that all functionalities of the module class are working as expected?

Reply:
Yes @suthagar23 the module class is working as intended 😄.